app/services/store_policy_resolver.py

```
@@ -0,0 +1,641 @@
+
+    @staticmethod
+    def _parse_bool_with_default(value: Any, default: bool) -> bool:
+        """Parse optional boolean field with default."""
+        if value is None:
+            return default
+        if not isinstance(value, bool):
+            raise PolicyParseError(f"Boolean field must be boolean, got {type(value).__name__}")
+        return value
+
+    @staticmethod
+    def _parse_decimal(value: Any, field_name: str) -> Decimal:
```

Ruff (0.16.0)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)